### PR TITLE
Switch to turn off soil erosion in CAM for dust emissions

### LINKED ISSUE #141: MAM5 and updates to dust emis
Add a mode for stratospheric sulfate to the modal aerosol model.
Updates to dust emissions.

Francis, this is my F90 codes and BLD codes to setup MAM5 configuration

/glade/u/home/zke3/scratch/cases/edit8

The F90 codes and chem_mech.in are in CAM, configure codes are in BLD folder, chem_mech.in and namelist files for WACCM are in WACCM folder. The F90 files are used for both WACCM and CAM. Two configurations I added in is trop_mam5 and waccm_tsmlt_mam5. If you need the codes to separate with the convection changes, I also have that older version. That is based on “release-cesm2.1.2” version.

Thanks 

Ziming Ke

Francis, here is our code changes,

/glade/u/home/longei/CAM-MAM5-codes/wet_removal_n02_cam6_2_020

 

and a list of the routines we modified.

DUSTMod.F90

dust_model.F90

modal_aer_opt.F90

Thanks very much.

All the best & Be well,

Longlei

### LINKED ISSUE #651: Move the dust emission source function and global tuning factor from CAM to CTSM
### Issue Type

Code Clean-up

### Issue Description

Hi this is Danny, a PhD student at UCLA and NCAR working with @jfkok and @dlawrenncar on CTSM dust modeling development. Here we request to move the CAM dust soil erodibility map and global tuning factor for the dust scheme previously read in and manipulated in CAM to CTSM. This issue is also posted on CTSM github https://github.com/ESCOMP/CTSM/issues/1836.

In issue https://github.com/ESCOMP/CTSM/issues/1604 we proposed to add a new dust emission scheme improving the dust mobilization parameterizations in CLM. Although a new scheme will be added as in PR https://github.com/ESCOMP/CTSM/pull/1712, CLM people decided to keep the existing emission scheme by Zender (2003). One issue is that the (time-independent) soil erodibility source function employed by Zender's scheme has always been read in and applied to the CAM emission flux instead of to the CLM emission flux, likely due to historical reasons. However, conceptually, how soil erodibility and land-surface characteristics affect the emission strength should be part of the land processes, and we believe it should be read in and applied in CLM instead of CAM. CLM has a variable for the Zender source function called mbl_bsn_fct, but it has always been set as 1 as a place holder and it is conceptually the same variable as the soil_erod in CAM. Here we request moving the soil erodibility map from CAM to CLM into mbl_bsn_fct, coming as part of either the surface dataset or the stream file. 

This small modification will make the CLM dust emission scheme more conceptually complete, and eliminate conceptually overlapping variables among CLM and CAM. 

Also, there is a global tuning factor (with CAM namelist variable dust_emis_fact) that controls the magnitude of the global total dust emission. Due to the same reason we think it should also be moved to CLM.

In CAM, we would like to modify the atm dust module and the namelist definition/defaults. Several Fortran variables, namelist variables, and output fields are proposed to be removed.

@dmleung proposed the modification. @dlawrenncar, @lkemmons, @L3atm, and software engineers including @ekluzek and @fvitt agreed about the decision.
@dmleung will work on this issue in both CLM and CAM with help from @ekluzek and @fvitt. 

I know CAM people will tune aerosols (including dust) often when they have new CAM versions. They might need to hereafter tune it in CLM instead of CAM. @cecilehannay is informed.

Thank you,
Danny

Files to be changed:
clm/src/biogeochem/DUSTMod.F90
clm/bld/namelist_files/namelist_defaults_ctsm.xml
clm/bld/namelist_files/namelist_definition_ctsm.xml
cam/bld/namelist_files/namelist_defaults_cam.xml
cam/bld/namelist_files/namelist_definitions.xml
cam/src/chemistry/modal_aero/dust_model.F90
cam/src/chemistry/aerosol/soil_erod_mod.F90
Some of the below CLM files will also be changed for adding the namelist variables and reading in the soil erodibility dataset:
clm/src/biogeophys/SoilStateType.F90
clm/src/biogeophys/SoilStateInitTimeConstMod.F90
clm/src/main/controlMod.F90
clm/src/main/clm_varctl.F90

P.S.

1. Subsequently, some CAM output fields, e.g., LND_MBL (Zender's source function) and DSTSFMBL (emission flux) will be cleared up. For instance, LND_MBL might become a CLM output instead of CAM output.
2. Path of the original soil erodibility data read in by CAM: soil_erod_file='/glade/p/cesmdata/cseg/inputdata/atm/cam/dst/dst_source2x2tunedcam6-2x2-04062017.nc'
3. The CAM soil erodibility file is 2x2 and is interpolated in CAM using the function lininterp to other resolutions. CLM will thus need to do the same (a stream file might be more capable of performing the interpolation).

### Will this change answers?

Yes

### Will you be implementing this yourself?

Yes

### PR DESCRIPTION
**Pull request by fvitt**
closes #651 
closes #141

### CODE REVIEW

**Comment by fvitt** [comment]
@ekluzek @dmleung This is the PR for soil erosion switch in CAM.  Please review.  Thanks.

File: bld/namelist_files/namelist_definition.xml [hunk 1]
```diff
@@ -6742,6 +6742,13 @@ Full pathname of boundary dataset for soil erodibility factors.
 Default: set by build-namelist.
 </entry>
 
+<entry id="soil_erod_active" type="logical"  category="cam_chem"
+       group="dust_nl" valid_values="" >
+Switch to turn on/off soil erodibility for dust emissions in CAM.
+Default: FALSE
```
Comment (ekluzek) [inline bld/namelist_files/namelist_definition.xml:6748]: Default should be TRUE, until after this is in CTSM and always on in there.

**Review by ekluzek — CHANGES_REQUESTED** [review]
The default for soil_erod_active should be TRUE, until we bring the changes into the CTSM side, when it'll be FALSE. Looks like this is only documented wrong in the namelist definition file.

**Comment by dmleung** [comment]
ESCOMP/CTSM#1967 now has the corresponding changes in CTSM.

**Comment by ekluzek** [comment]
This also relates to #778.

This will also update both CMEPS and CTSM, so will happen after those tags are made.

The CMEPS issue is: https://github.com/ESCOMP/CMEPS/issues/353

**Comment by ekluzek** [comment]
Added a design doc about this here:

https://docs.google.com/document/d/18nZ3LJF5W-YF9iBhqed6s_NWeKOvSSL2-k0Lye1nnLg

**Comment by ekluzek** [comment]
When we are closer to having this all together and working, we should validate that the dust emission is similar when in CTSM as in CAM by comparing results between the two. We expect answers to be different by greater than roundoff, but they should be qualitatively similar.

**Comment by ekluzek** [comment]
@fvitt note that the CMEPS PR has been tagged as cmeps0.14.73. The PR to use that in CTSM is https://github.com/ESCOMP/CTSM/pull/2545. Once the CTSM tag is in place this one could come in.

**Comment by ekluzek** [comment]
The latest CAM (cam6_4_015) has already updated beyond the needed CMEPS tag (to cmeps1.0.2), so this can start to be changed to use the namelist items in the drv_flds_in for dust_emis_inparm. If this PR is hard to update, it could also be restarted from the latest CAM tag.

**Comment by fvitt** [comment]
@ekluzek Thanks for the comments and pointing me to this PR.   I will probably start a new branch and PR this is quite old...

**Pull request closed without merging**